src/Carousel.js

```diff
@@ -280,7 +280,7 @@ const Carousel = React.createClass({
       direction = direction || this.getDirection(previousActiveIndex, index);
 
       if (this.props.onSelect) {
-        this.props.onSelect(index, direction);
+        this.props.onSelect(index, e, direction);
```

Comment:
I am wondering if we should make that second argument like an object ball expando, so we can add stuff on to it indefinately?